lib/cli/writer.js

```diff
@@ -67,6 +68,13 @@ exports.writeImage = (imagePath, drive, options, onProgress) => {
     return fs.openAsync(drive.raw, 'rs+');
   }).then((driveFileDescriptor) => {
     return imageStream.getFromFilePath(imagePath).then((image) => {
+      if (!image.size.final.estimation && image.size.final.value > drive.size) {
```

Comment:
It took quite a while, but I was able to normalise these. The resulting refactoring PR is far from trivial, so can we approve and merge this one, and I'll send the refactor PR right afterwards?